# Advancing microRNA Target Site Identification via Bias-Corrected Chimeric Datasets for Machine Learning Approaches

This repository contains all the code required to reproduce the work presented for my dissertation, submitted in partial fulfilment of the requirements for the degree of M.Sc. in Molecular Medicine (By Research) in July 2025. 

The directories are ordered chronologically, corresponding to the order of the analyses and experiments presented in the dissertation. Each analysis/experiment is designed to be run independently. In general, subsequent analyses build upon the outputs or results generated by preceding ones.

Each directory follows the same structure, as follows:
```
project-root/
│
├── 00_first_analysis/
│   ├── code/
│   ├── data/
│   ├── results/
│   ├── RUNME.sh
│   └── README.md
│
├── 01_second_analysis/
│   ├── code/
│   ├── data/
│   ├── results/
│   ├── RUNME.sh
│   └── README.md
│
└── README.md           # This file
```
The `README.md` provides a description of the analysis or experiment and instructions on how to run it in the respective directory. It generally follows a common structure, including a description, list of dependencies (including versions used), and further instructions or notes where necessary. 

The `RUNME.sh` master script is an executable file for running the entire analysis or experiment. It executes other scripts from the `code/` directory as needed. 

The `code/` directory contains Python, R or bash scripts needed to run the analysis or experiment. Each script contains docstrings including a description, usage, and arguments. These scripts are executed by the respective `RUNME.sh` master script. 

Unless indicated otherwise in the respective `README.md`, the `RUNME.sh` master script should be run *from the parent directory of the analysis/experiment* as follows: 
```bash
bash RUNME.sh
``` 
Most master scripts contain SBATCH directives (edit as necessary!) and can run on a HPC with SLURM using: 
```bash
sbatch RUNME.sh
```

The `data/` and `results/` directories are placeholder input and output directories, respectively. These are created when the `RUNME.sh` master script is executed. 

A `RUNME.log` file is outputted to the `results/` directory. 

## Publications

**Most of the work presented here has been included in the following publication:**

Sammut, S., Gresova, K., Tzimotoudis, D., Marsalkova, E., Cechak, D., & Alexiou, P. (2025b). miRBench: Novel benchmark datasets for microRNA binding site prediction that mitigate against prevalent microRNA frequency class bias. Bioinformatics, 41, i542–i551. https://doi.org/10.1093/bioinformatics/btaf233

**Other publications most relevant to this work include:**

Hejret, V., Varadarajan, N. M., Klimentova, E., Gresova, K., Giassa, I.-C., Vanacova, S., & Alexiou, P. (2023). Analysis of chimeric reads characterises the diverse targetome of AGO2-mediated regulation. Scientific Reports, 13. https://doi.org/10.1038/s41598-023-49757-z

Klimentová, E., Hejret, V., Krčmář, J., Grešová, K., Giassa, I.-C., & Alexiou, P. (2022). miRBind: A deep learning method for miRNA binding classification. Genes, 13(12). https://doi.org/10.3390/genes13122323

Manakov, S. A., Shishkin, A. A., Yee, B. A., Shen, K. A., Cox, D. C., Park, S. S., Foster, H. M., Chapman, K. B., Yeo, G. W., & Van Nostrand, E. L. (2022). Scalable and deep profiling of mRNA targets for individual microRNAs with chimeric eCLIP. bioRxiv. https://doi.org/10.1101/2022.02.13.480296
